server/handler/status.go

```diff
@@ -31,17 +33,29 @@ type Status struct {
 
 func (h *Status) Handles() []string { return []string{"status"} }
 
-// Handle pull_request
+// Handle status
 // https://developer.github.com/v3/activity/events/types/#statusevent
 func (h *Status) Handle(ctx context.Context, eventType, deliveryID string, payload []byte) error {
 	var event github.StatusEvent
 	if err := json.Unmarshal(payload, &event); err != nil {
 		return errors.Wrap(err, "failed to parse status event payload")
 	}
 
+	if !strings.HasPrefix(event.GetContext(), h.PullOpts.StatusCheckContext) {
```

Comment:
Given the introduction of `processOthers` and `processOwn`, I think it would be better to invert this condition: if the context has the prefix, call `processOwn` and return.